napari/layers/labels/labels.py

```diff
@@ -282,6 +287,30 @@ def properties(self, properties: Dict[str, np.ndarray]):
         self._properties = self._validate_properties(properties)
         self._label_index = label_index
 
+    @property
+    def color_dict(self):
+        """dict: custom color dict for label coloring"""
+        return self._color_dict
+
+    @color_dict.setter
+    def color_dict(self, color_dict):
+
+        if not color_dict:
+            self._color_dict = {
+                self._background_label: 'transparent',
+                None: 'black',
+            }
+            self.color_mode = ColorMode.DIRECT
+        else:
+            if self._background_label not in color_dict:
+                color_dict[self._background_label] = 'transparent'
+
+            if None not in color_dict:
+                color_dict[None] = 'black'
+
```

Comment:
I'd probably move the `transform_color(color_str)[0]` in here and out of `color_dict_to_colormap` too so that we know that each color is valid and in rgba form. I think this will make things like serialization and the `_get_state` call nicer because we'll know with more confidence exactly what goes in there. Thoughts @kevinyamauchi?

Reply:
We should add some docs around color handling within napari @kevinyamauchi, right now there isn't any. The rationale was that by having everything converted to an RGBA representation there would then be one known way that other tools could interact with colors from napari. For example, one day if a plugin were to be passed colors via napari that plugin could assume that the color was an RGBA representation instead of a string like `lime-green` which that plugin might not know how to deal with. Putting the call inside the labels layer ensures that we'll achieve that uniformity here. Does that make sense?

Reply:
I managed to decouple the color dict and color mode, and removed the init from here

Reply:
it does, refactored